---
# the default layout is 'page'
icon: fas fa-info-circle
order: 4
---

# Pinyuan Li

Machine Learning Researcher & Software Developer @ Emory

ðŸ“§ [EmoryPatrick@outlook.com](mailto:EmoryPatrick@outlook.com)
ðŸ”— [LinkedIn](https://www.linkedin.com/in/patrick-pinyuan-li)
ðŸ–¥ [GitHub](https://github.com/BrandeisPatrick)

## Skills

- BERT, Large Language Model (LLM), Diffusion Models, Contrastive Learning
- Python, Java, C
- Software Development, Algorithm Design, Statistical Modeling
- Data Visualization, Computer Cluster Management, Parallel Computing
- Job Scheduling with SLURM, Google Cloud Platform (GCP), Amazon Web Services (AWS)
- Weights & Biases, TensorBoard, Scikit-Learn

## Work Experience

### Emory University, Bromberg Lab â€” Machine Learning Researcher
*Jun 2024 â€“ Present*
- Led a National Science Foundation-funded AI4Science project, developing reliable machine-learning solutions for complex biological problems.
- Built scalable algorithms and implemented machine learning models to help LLMs better understand biology domain data.
- Worked with fellow researchers to improve explainable AI for foundational biological models, including AlphaFold by analyzing self-attention maps.

### Emory University, Fang Lab â€” Machine Learning Researcher
*Sep 2021 â€“ Present*
- Collaborated with a multidisciplinary team to design and implement machine learning solutions that utilize chemical features, resulting in improved model performance for chemistry applications.
- Developed a curated dataset of 10,000 chemical compounds with handcrafted features, optimized for AI-driven chemistry models. It received 10 citations within the first year.

### Brandeis University, Herzfeld Lab â€” Undergraduate Software Developer
*Aug 2019 â€“ Aug 2021*
- Co-developed LEWIS, a semi-classical force field simulation tool in C, enabling accurate simulation and prediction of molecular dynamics.
- Managed the complete software lifecycleâ€”including development, deployment, maintenance, and updatesâ€”to ensure consistent performance and reliability.

## Selected Projects

### Protein Siamese Neural Network: Predicting Single Nucleotide Variant Effects
- Engineered and optimized deep learning models using contrastive learning for high-performance text semantic similarity detection for protein sequence data.
- Built a data pipeline to process a 100K-entry dataset efficiently.
- Developed a classification model inspired by NLP-based text similarity techniques, improving mutation effect prediction accuracy by 7% over the baseline model.
- Developed automated training pipelines with experiment tracking and gradient visualization.
- Published well-documented, modular code on GitHub, ensuring collaborative development and maintainability.

### Learning Protein-ish: Foundational Insights on Protein Language Models
- Designed and deployed a domain-specific LLM, leveraging Hugging Face, PyTorch, and scikit-learn, optimizing biological sequence interpretation.
- Built a pipeline to scrape, clean, and filter a 100K-entry dataset from UniProt using Python and SQL, refining it into a high-quality 10K dataset.
- Conducted an architectural analysis of state-of-the-art LLMs, evaluating their efficiency, scalability, and real-world applications.
- Used adversarial probing techniques to analyze attention mechanisms in LLMs, improving interpretability for protein-sequence prediction.
- Benchmarked models using accuracy, F1-score, and AUC-ROC, optimizing for performance and efficiency.
- Deployed the LLM on Google Cloud Platform under the Google Cloud Research Credits Program, ensuring scalability and cloud-based accessibility.

### LEWIS: Semi-Classical Molecular Force Field Software
- Developed a high-performance molecular simulation engine in C, optimizing semi-classical force field parameters for enhanced computational chemistry.
- Implemented Differential Evolution and Monte Carlo methods for parameter optimization, achieving a 50% improvement in computational speed over state-of-the-art approaches.

## Education

- **Emory University**, Masterâ€™s in Computational Chemistry (Sep 2021 â€“ Present)  
  Co-advisors: Fang Liu (Chemistry), Yana Bromberg (Biology), and Joyce Ho (Computer Science)
- **Brandeis University**, Bachelor of Science in Biochemistry (Jan 2019 â€“ May 2021)  
  Transferred from Michigan State University in 2019
- **Michigan State University**, Bachelor of Science in Physics (Sep 2017 â€“ Jan 2019)

## Awards

Brandeis Provostâ€™s Undergraduate Research Fellowship, Google Cloud Research Credits Recipient, Michigan State Honors College, Michigan State Advanced Mathematics Track Program.

